developersample/src/main/java/com/microsoft/identity/client/developersample/AuthUtil.java

```diff
@@ -0,0 +1,215 @@
+    private static final String[] SCOPES = {"https://graph.microsoft.com/User.Read"};
+    private static final String[] EXTRA_SCOPES = {"Calendars.Read"};
+    private static final String CLIENT_ID = "9851987a-55e5-46e2-8d70-75f8dc060f21";
+    private List<User> mUsers = null;
+    final Activity mActivity;
+
+    /**
+     * When initializing the {@link PublicClientApplication}, all the apps should only provide us the application context instead of
+     * the running activity itself. If running activity itself is provided, that will have the sdk hold a strong reference of the activity
+     * which could potentially cause the object not correctly garbage collected and cause activity leak.
+     * <p>
+     * External Logger should be provided by the Calling app. The sdk logs to the logcat by default, and loglevel is enabled at verbose level.
+     * To set external logger,
+     * {@link Logger#setExternalLogger(ILoggerCallback)}.
+     * To set log level,
+     * {@link Logger#setLogLevel(Logger.LogLevel)}
+     * By default, the sdk won't give back any Pii logging. However the app can turn it on, this is up to the application's privacy policy.
+     * To turn on the Pii logging,
+     * {@link Logger#setEnablePII(boolean)}
+     * Application can also set the component name. There are cases that other sdks will also take dependency on MSAL i.e. microsoft graph sdk,
+     * providing the component name will help separate the logs from application and the logs from the sdk running inside of
+     * the apps.
+     * To set component name:
+     * {@link PublicClientApplication#setComponent(String)}
+     * <p>
+     * For the {@link AuthenticationCallback}, MSAL exposes three results 1) Success, which contains the {@link AuthenticationResult} 2) Failure case,
+     * which contains {@link MsalException} and 3) Cancel, specifically for user canceling the flow.
+     * <p>
+     * For the failure case, MSAL exposes three sub exceptions:
+     * 1) {@link MsalClientException}, which is specifically for the exceptions running inside the client app itself, could be no active network,
+     * Json parsing failure, etc.
+     * 2) {@link MsalServiceException}, which is the error that the sdk gets back when communicating to the service, could be oauth2 errors, socket timout
+     * or 500/503/504. For oauth2 erros, MSAL returns back the exact error that server returns back to the sdk.
+     * 3) {@link MsalUiRequiredException}, which means that UI is required.
+     */
+    public AuthUtil(final Activity activity) {
+        if (mApplication == null) {
+            mApplication = new PublicClientApplication(activity.getApplicationContext(), CLIENT_ID);
+        }
+        mActivity = activity;
+    }
+
+    // Get the number of users signed into the app. In the context of the current app this method can return 1 or 0
+    int getUserCount() {
+        getUsers();
+        if (mUsers == null) {
```

Comment:
I don't think i'm following, same as what I mentioned above, do a addAll, if getusers returns you somethings, the list will contain users. If not, it will still be an empty list.